### ISSUE TITLE
[index] Using "type!POD" vs. "POD type"

### ISSUE THREAD

**Issue by jlaire** [cplusplus/draft#1033]
There are index entries for "*callable type*", "*integer type*", "*object type*", "*scalar type*" and many others of this form. The index also has a list of terms under *type*, such as "*type!POD*" and "*type!function*". A few appear in both forms: there's "*integral type*" as well as "*type!integral*".

This makes the index hard to use, because it's not clear (to me) which form is used for any given term. I noticed this when trying to find the definition of "POD type". Would it make sense to list all of these terms as both "*foo type*" and "*type!foo*"?

**Comment by jensmaurer** [comment]
We do say "foo type, see type, foo" in similar cases (maybe not with "type") and keep the list of page numbers exclusively with "type, foo".

**Comment by burblebee** [comment]
Please document the preferences/guidelines somewhere for future editor's benefit.

**Comment by jensmaurer** [comment]
In the index of library names, maybe distinguish class names from functions etc, e.g. postfix with ::. Alisdair dislikes :: not connecting things. Different uses of the index: One, just lookup a name (hierarchy not useful). Two, as a reference guide to the library names including nesting and subnamespaces.

**Comment by AlisdairM** [comment]
What do we believe remains to be addressed after landing #2258?
Or are we ready to close this issue?

**Comment by frederick-vs-ja** [comment]
We haven't indexed _POD type_ with `\defnadj{POD}{type}`, possibly because it was deprecated. But I still think we should index the term with `\defnadj`.
https://github.com/cplusplus/draft/blob/c59d8c8e9d6fe8adba2142d1344b0e142c9d6ff4/source/future.tex#L320-L325

**Comment by jensmaurer** [comment]
Yes, I think we should replace the quoted \term things with \defnadj .
